tests/boards/intel_s1000_crb/main/boards.s1000_crb.main fails to build

Running:
`./scripts/sanitycheck -p intel_s1000_crb -s tests/boards/intel_s1000_crb/main/boards.s1000_crb.main`

Gets:
```
[ 74%] Building C object zephyr/drivers/i2s/CMakeFiles/drivers__i2s.dir/i2s_cavs.c.obj
In file included from /home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:19:
/home/galak/git/zephyr/include/device.h:153:26: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  153 | #define DEVICE_GET(name) (&DEVICE_NAME_GET(name))
      |                          ^
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:56:3: note: in expansion of macro 'DEVICE_GET'
   56 |   DEVICE_GET(I2S_DEVICE_NAME(i2s_id))
      |   ^~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:66:18: note: in expansion of macro 'I2S_DEVICE_OBJECT'
   66 |     .user_data = I2S_DEVICE_OBJECT(i2s_id),\
      |                  ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:890:1: note: in expansion of macro 'I2S_DEVICE_DATA_DEFINE'
  890 | I2S_DEVICE_DATA_DEFINE(1);
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:153:26: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  153 | #define DEVICE_GET(name) (&DEVICE_NAME_GET(name))
      |                          ^
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:56:3: note: in expansion of macro 'DEVICE_GET'
   56 |   DEVICE_GET(I2S_DEVICE_NAME(i2s_id))
      |   ^~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:82:18: note: in expansion of macro 'I2S_DEVICE_OBJECT'
   82 |     .user_data = I2S_DEVICE_OBJECT(i2s_id),\
      |                  ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:890:1: note: in expansion of macro 'I2S_DEVICE_DATA_DEFINE'
  890 | I2S_DEVICE_DATA_DEFINE(1);
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:153:26: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  153 | #define DEVICE_GET(name) (&DEVICE_NAME_GET(name))
      |                          ^
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:56:3: note: in expansion of macro 'DEVICE_GET'
   56 |   DEVICE_GET(I2S_DEVICE_NAME(i2s_id))
      |   ^~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:66:18: note: in expansion of macro 'I2S_DEVICE_OBJECT'
   66 |     .user_data = I2S_DEVICE_OBJECT(i2s_id),\
      |                  ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:899:1: note: in expansion of macro 'I2S_DEVICE_DATA_DEFINE'
  899 | I2S_DEVICE_DATA_DEFINE(2);
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:153:26: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  153 | #define DEVICE_GET(name) (&DEVICE_NAME_GET(name))
      |                          ^
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:56:3: note: in expansion of macro 'DEVICE_GET'
   56 |   DEVICE_GET(I2S_DEVICE_NAME(i2s_id))
      |   ^~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:82:18: note: in expansion of macro 'I2S_DEVICE_OBJECT'
   82 |     .user_data = I2S_DEVICE_OBJECT(i2s_id),\
      |                  ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:899:1: note: in expansion of macro 'I2S_DEVICE_DATA_DEFINE'
  899 | I2S_DEVICE_DATA_DEFINE(2);
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:153:26: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  153 | #define DEVICE_GET(name) (&DEVICE_NAME_GET(name))
      |                          ^
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:56:3: note: in expansion of macro 'DEVICE_GET'
   56 |   DEVICE_GET(I2S_DEVICE_NAME(i2s_id))
      |   ^~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:66:18: note: in expansion of macro 'I2S_DEVICE_OBJECT'
   66 |     .user_data = I2S_DEVICE_OBJECT(i2s_id),\
      |                  ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:908:1: note: in expansion of macro 'I2S_DEVICE_DATA_DEFINE'
  908 | I2S_DEVICE_DATA_DEFINE(3);
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/device.h:153:26: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  153 | #define DEVICE_GET(name) (&DEVICE_NAME_GET(name))
      |                          ^
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:56:3: note: in expansion of macro 'DEVICE_GET'
   56 |   DEVICE_GET(I2S_DEVICE_NAME(i2s_id))
      |   ^~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:82:18: note: in expansion of macro 'I2S_DEVICE_OBJECT'
   82 |     .user_data = I2S_DEVICE_OBJECT(i2s_id),\
      |                  ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/i2s/i2s_cavs.c:908:1: note: in expansion of macro 'I2S_DEVICE_DATA_DEFINE'
  908 | I2S_DEVICE_DATA_DEFINE(3);
      | ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/drivers/i2s/i2s_cavs.c
+++ b/drivers/i2s/i2s_cavs.c
@@ -63,7 +63,7 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 				.source_burst_length = CAVS_I2S_DMA_BURST_SIZE,\
 				.dest_burst_length = CAVS_I2S_DMA_BURST_SIZE,\
 				.dma_callback = i2s_dma_tx_callback,	\
-				.user_data = I2S_DEVICE_OBJECT(i2s_id),\
+				.user_data = (void *)I2S_DEVICE_OBJECT(i2s_id),\
 				.complete_callback_en = 1,	\
 				.error_callback_en = 1,		\
 				.block_count = 1,		\
@@ -79,7 +79,7 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 				.source_burst_length = CAVS_I2S_DMA_BURST_SIZE,\
 				.dest_burst_length = CAVS_I2S_DMA_BURST_SIZE,\
 				.dma_callback = i2s_dma_rx_callback,\
-				.user_data = I2S_DEVICE_OBJECT(i2s_id),\
+				.user_data = (void *)I2S_DEVICE_OBJECT(i2s_id),\
 				.complete_callback_en = 1,	\
 				.error_callback_en = 1,		\
 				.block_count = 1,		\

